src/tactics/tactics_main.ml

```diff
@@ -18,8 +18,21 @@ let try_usolve (opts : s_opts) (lst : sopt_t list) (ret : s_opts -> unit Proofvi
         end
   end
 
-let use_lemmas lst =
+let with_delayed_uconstrs ist cs tac =
+  let flags = {
+    Pretyping.use_typeclasses = Pretyping.UseTC;
+    solve_unification_constraints = true;
+    fail_evar = false;
+    expand_evars = true;
+    program_mode = false;
+    polymorphic = false;
+  } in
+  let cs = List.map (Tacinterp.type_uconstr ~flags ist) cs in
+  Tacticals.New.tclMAPDELAYEDWITHHOLES true cs tac
```

Comment:
Tacticals.New.tclMAPDELAYEDWITHHOLES is available only since Coq 8.12.2. It would be nice to avoid it, to keep compatible with earlier versions.

Reply:
Thanks for the review. Originally I did use another API, but I changed it to this one since it seems cleaner. Didn't know this is a new one. I will try fix it soon.